src/ucp/rndv/proto_rndv.c

```diff
@@ -365,41 +348,136 @@ ucp_proto_rndv_ctrl_init(const ucp_proto_rndv_ctrl_init_params_t *params,
         ucp_proto_perf_range_add_data(&remote_perf);
         ucp_proto_perf_node_add_child(remote_perf.node, remote_range->node);
 
-        parallel_stages[0] = &ctrl_perf;
+        parallel_stages[0] = ctrl_range;
         parallel_stages[1] = &remote_perf;
         status = ucp_proto_init_parallel_stages(params->ctrl_msg_name,
                                                 min_length, range_max_length,
                                                 params->perf_bias,
                                                 parallel_stages, 2, proto_caps);
         if (status != UCS_OK) {
-            goto out_deref_perf_node;
+            return status;
         }
 
         ucp_proto_perf_node_deref(&remote_perf.node);
 
-        min_length = range_max_length - 1;
+        min_length = range_max_length + 1;
     } while ((remote_range++)->max_length < max_length);
 
-    status = UCS_OK;
+    return UCS_OK;
+}
 
-out_deref_perf_node:
-    ucp_proto_perf_node_deref(&ctrl_perf.node);
-out:
-    return status;
+void
+ucp_proto_rndv_set_variant_config(const ucp_proto_init_params_t *init_params,
```

Comment:
why do we need to initialize a ucp_proto_config_t struct?

Reply:
For `query()` interface calls in case of RTS remote proto initialization and for both `query()` and `progress()` in case of RNDV PPLN underlying proto initialization.